devito/expression.py

```diff
@@ -18,6 +26,9 @@ def __init__(self, stencil):
         self.functions = []
         # Traverse stencil to determine meta information
         for e in preorder_traversal(self.stencil):
+            if isinstance(e, SymbolicData):
```

Comment:
*Not* as part of this PR and I know this is me being picky:
as of sympy 1.0.0, all sympy objects have an ``is_Type`` property to find out what they are. Even Indexed and IndexedBase are added this property. Maybe we should extend our hierarchy to do the same? eg is_SymbolicData?